lib/libzfs/libzfs_sendrecv.c

```diff
@@ -4014,6 +4006,23 @@ zfs_setup_cmdline_props(libzfs_handle_t *hdl, zfs_type_t type,
 				fnvlist_add_nvpair(*oxprops, nvp);
 			break;
 		case DATA_TYPE_STRING: /* -o property=value */
+		/*
+		 * we're trying to override a property that does not
+		 * make sense for this type of dataset, but we don't want to
+		 * fail if the receive is recursive: this comes in handy when
+		 * the send stream contains, for instance, a child ZVOL and
+		 * we're trying to receive it with "-o atime=on"
+		 */
```

Comment:
nit: cstyle this comment appears to be indented to the wrong level.

Reply:
Thanks, resolved and squashed.
Sidenote: `make checkstyle` (not cstyle) spams a lot(!) of 
```
awk: cmd. line:1: warning: regexp escape sequence `\!' is not a known regexp operator
awk: cmd. line:1: warning: regexp escape sequence `\#' is not a known regexp operator
```
 (for me?)

Reply:
>  Sidenote: make checkstyle (not cstyle) spams

I've opened #11902 to resolve this.